### PR TITLE
Get product by group

### PR DESCRIPTION
**Pull request by padovese**

### CODE REVIEW

File: src/main/kotlin/com/maitre/menuservice/adapter/product/in/get/GetProductByGroupHandler.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.maitre.menuservice.adapter.product.`in`.get
+
+import org.springframework.web.reactive.function.server.ServerRequest
+import org.springframework.web.reactive.function.server.ServerResponse
+import reactor.core.publisher.Mono
+
+interface GetProductByGroupHandler {
```
Comment (diegolucasilva) [inline src/main/kotlin/com/maitre/menuservice/adapter/product/in/get/GetProductByGroupHandler.kt]: Products*

File: src/main/kotlin/com/maitre/menuservice/adapter/product/in/get/GetProductByGroupHandlerImpl.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+@Component
+class GetProductByGroupHandlerImpl(
+    private val logger: Logger,
+    private val getProductByGroupUseCase: GetProductByGroupUseCase
+) : GetProductByGroupHandler {
+
+    override fun getByGroup(serverRequest: ServerRequest): Mono<ServerResponse> {
+        logger.info(
+            "Get Product By Group Request initiated. " +
```
Comment (diegolucasilva) [inline src/main/kotlin/com/maitre/menuservice/adapter/product/in/get/GetProductByGroupHandlerImpl.kt]: Products*

File: src/main/kotlin/com/maitre/menuservice/adapter/product/in/get/GetProductByGroupHandlerImpl.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.web.reactive.function.server.ServerResponse.ok
+import reactor.core.publisher.Mono
+
+@Component
+class GetProductByGroupHandlerImpl(
+    private val logger: Logger,
+    private val getProductByGroupUseCase: GetProductByGroupUseCase
+) : GetProductByGroupHandler {
+
+    override fun getByGroup(serverRequest: ServerRequest): Mono<ServerResponse> {
```
Comment (diegolucasilva) [inline src/main/kotlin/com/maitre/menuservice/adapter/product/in/get/GetProductByGroupHandlerImpl.kt]: the method name can be execute()

File: src/main/kotlin/com/maitre/menuservice/domain/product/usecase/GetProductByGroupUseCase.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import com.maitre.menuservice.domain.product.entity.Product
+import com.maitre.menuservice.domain.product.port.out.persistence.GetProductByGroupPort
+import com.maitre.menuservice.exception.ProductNotFoundException
+import org.slf4j.Logger
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+
+@Service
+class GetProductByGroupUseCase(
```
Comment (diegolucasilva) [inline src/main/kotlin/com/maitre/menuservice/domain/product/usecase/GetProductByGroupUseCase.kt]: Products*

**Comment by padovese** [comment]
done.

**Pull request merged**